contracts/protocol/libraries/logic/SupplyLogic.sol

```diff
@@ -203,12 +202,11 @@ library SupplyLogic {
             DataTypes.AssetType.ERC721
         );
 
-        bool isFirstSupply = INToken(reserveCache.xTokenAddress).mint(
+        bool isFirstCollateral = INToken(reserveCache.xTokenAddress).mint(
             params.onBehalfOf,
             params.tokenData
         );
-        // TODO consider using (usedAsCollateral > 0) instead here to enable collateralization
-        if (isFirstSupply) {
+        if (isFirstCollateral) {
```

Comment:
`oldCollateralBalance == 0 && newCollateralBalance > oldCollateralBalance`

Reply:
same as above. `isFirstCollateral` is nice and simpler.

Reply:
but it's more general to handle more cases. You can also easily wrap it inside a helper function